### DISCUSSION TITLE
Duplicate log handlers causing duplicate logs

### DISCUSSION THREAD

**Discussion by J0hn-Lawrence** [chaitu-ycr/py_canoe#37] [Q&A] [answered]
Hey there, 
first of all thank you @chaitu-ycr for making a great library, your efforts are greatly appreciated! 

My observation:
I am tinkering around with the library recently and I noticed that I get duplicate/multiple logs, which is due to duplicate/multiple log handlers (confirmed it in the constructor of the PyCanoeLogger class, see how below). These cause the same log message to be displayed multiple times, which I think is a bit annoying. 

My question: 
Maybe I am using the library incorrectly, thus causing this problem. Has anybody else encountered this or a similar problem so far?

My (temporary) solution: 
I added a check/removal to the last line of the PyCanoeLogger's __init__() which looks like this:
```
while len(self.log.handlers) > 1:
    self.log.removeHandler(self.log.handlers[0])
```

I did not create an issue for this topic, as this may be caused by misuse of the library from my part. If this Q&A section is the wrong spot for this inquiry, just let me know and I will move it elsewhere.

Appreciate any feedback from you guys. Have a great week!

**Comment by chaitu-ycr** [comment]
hi @J0hn-Lawrence 
in latest release iam clearing log handlers before init.
with this change in my setup i am not observing duplicate log files.
can you also please check and close if this issue resolved.

https://github.com/chaitu-ycr/py_canoe/tree/v2.1.0
https://github.com/chaitu-ycr/py_canoe/blob/v2.1.0/src/py_canoe_utils/py_canoe_logger.py#L11

**Comment by J0hn-Lawrence** [comment, marked as answer]
Hey @chaitu-ycr

Thanks for the feedback.

I can confirm that the new release 2.1.0 indeed fixes the issue of duplicate logs. Thank you for your work!

In hindsight I should have probably created an issue for this instead of a discussion, sorry for that.

I will close this discussion. Have a nice week.
